spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpInboundGateway.java

```diff
@@ -117,7 +139,7 @@ public boolean onMessage(Message<?> message) {
 	}
 
 	private boolean doOnMessage(Message<?> message) {
-		Message<?> reply = this.sendAndReceiveMessage(message);
+		Message<?> reply = sendAndReceiveMessage(message);
```

Comment:
Need to catch and swallow `MessageTimeoutException` to retain existing behavior when no EC or error flow returns `null`.

Reply:
Well, what we have here is fully a new behavior which doesn't break an existing one by default because all the existing `TcpInboundGateways` are created without that new `errorOnTimeout` ctor.
Also see an explanation of that option in the super class:
```
  If no error channel is
	 * configured, the {@link MessageTimeoutException} will be thrown.
```

So, what you are asking is fully redundant since it is related only to new option and we definitely should throw that `MessageTimeoutException` according contact. And again: we do that only if `errorOnTimeout` is turned on, which is not by default.

Does it make sense?

Reply:
Good point; I was concerned about generating ERROR logs in the caller to `onMessage` but, as you say, they have to opt-in by setting the boolean, in which case an error log is appropriate anyway.